tests/test_prod_smoketest.py

```diff
@@ -0,0 +1,203 @@
+#!/usr/bin/env python
+"""
+A prod test for the DSS, checks core API Requests to know whats available. Work with data present.
+"""
+import os, sys, argparse, time, uuid, json, shutil, tempfile, unittest
+import subprocess
+
+import boto3
+import botocore
+from cloud_blobstore import BlobStore
+from itertools import product
+
+pkg_root = os.path.abspath(os.path.join(os.path.dirname(__file__), ".."))  # noqa
+sys.path.insert(0, pkg_root)  # noqa
+
+from dss import Replica
+from tests.infra import testmode
+from tests import test_smoketest
+from dss.storage.checkout.bundle import get_dst_bundle_prefix
+
+parser = argparse.ArgumentParser(description=__doc__)
+parser.add_argument("--no-clean", dest="clean", action="store_false",
+                    help="Don't remove the temporary working directory on exit.")
+args = argparse.Namespace(clean=True)
+
+def GREEN(message=None):
+    if message is None:
+        return "\033[32m" if sys.stdout.isatty() else ""
+    else:
+        return GREEN() + str(message) + ENDC()
+
+def RED(message=None):
+    if message is None:
+        return "\033[31m" if sys.stdout.isatty() else ""
+    else:
+        return RED() + str(message) + ENDC()
+
+def ENDC():
+    return "\033[0m" if sys.stdout.isatty() else ""
+
+
+def run(command, **kwargs):
+    print(GREEN(command))
+    try:
+        return subprocess.run(command, check=True, shell=isinstance(command, str), **kwargs)
+    except subprocess.CalledProcessError as e:
+        parser.exit(RED(f'{parser.prog}: Exit status {e.returncode} while running "{command}". Stopping.'))
+
+
+def run_for_json(command, **kwargs):
+    return json.loads(run(command, stdout=subprocess.PIPE, **kwargs).stdout.decode(sys.stdout.encoding))
+
+
+@testmode.integration
+class ProdSmoketest(unittest.TestCase):
```

Comment:
This should inherit `Smoketest()` from `test/test_smoketest.py` and only change what's needed from there (and the functions above should be imported from `test/test_smoketest.py`).

Either that or modify `test/test_smoketest.py` to run on all stages as appropriate.

Reply:
created base class for regular + prod smoketest to inherit from.